docs/guides/kds-sync-name.md

```diff
@@ -0,0 +1,186 @@
+# What is the name of the resource after KDS sync?
+
+## Old policies (i.e. TrafficRoute, TrafficPermission)
+
+* CRD scope: Cluster
+* Kuma scope: Mesh
+
+### Global(k8s/universal) -> Zone(k8s/universal)
+
+```
+nameInZone := "${truncate(nameInGlobal, 236)}-${hash(mesh, nameInGlobal)}"
+```
+
+Example: `my-policy -> my-policy-fvdw2c2wx44fv4xz`
+
+### Zone(k8s/universal) -> Global(k8s/universal)
+
+Not supported
+
+## New policies (i.e. MeshTrafficPermission, MeshHTTPRoute)
+
+* CRD scope: Namespace
+* Kuma scope: Mesh
+
+### Global(k8s) -> Zone(k8s)
+
+```
+nameInGlobal = trimNsSuffix(nameInGlobal)
+nameInZone := "${truncate(nameInGlobal, 236)}-${hash(mesh, nameInGlobal)}.${systemNamespace}"
+```
+
+Example: `my-policy.kuma-system-global -> my-policy-fvdw2c2wx44fv4xz.kuma-system-zone`
+
+### Global(universal) -> Zone(k8s)
+
+```
+nameInZone := "${truncate(nameInGlobal, 236)}-${hash(mesh, nameInGlobal)}.${systemNamespace}"
+```
+
+Example: `my-policy -> my-policy-fvdw2c2wx44fv4xz.kuma-system-zone`
+
+### Global(k8s) -> Zone(universal)
+
+```
+nameInGlobal = trimNsSuffix(nameInGlobal)
+nameInZone := "${truncate(nameInGlobal, 236)}-${hash(mesh, nameInGlobal)}"
+```
+
+Example: `my-policy.kuma-system-global -> my-policy-fvdw2c2wx44fv4xz`
+
+### Global(universal) -> Zone(universal)
+
+```
+nameInZone := "${truncate(nameInGlobal, 236)}-${hash(mesh, nameInGlobal)}"
```

Comment:
Why doing truncation when doing universal --> universal ? I thought there wasn't a name length limit.

Reply:
No, we have a name length limit https://github.com/kumahq/kuma/blob/60733cb1f89c08de1f4e4fa25036ced4de2f2f5f/pkg/plugins/resources/postgres/migrations/data/1677096751_update_resources_name_limit.up.sql#L2